specification/dartLangSpec.tex

```diff
@@ -1850,20 +1851,21 @@ \section{Classes}
   \alt \STATIC{}? <setterSignature>
   \alt <operatorSignature>
 
-<declaration> ::= <constantConstructorSignature> (<redirection> | <initializers>)?
-  \alt <constructorSignature> (<redirection> | <initializers>)?
+<declaration> ::= \EXTERNAL{} <factoryConstructorSignature>
   \alt \EXTERNAL{} <constantConstructorSignature>
   \alt \EXTERNAL{} <constructorSignature>
   \alt (\EXTERNAL{} \STATIC{}?)? <getterSignature>
   \alt (\EXTERNAL{} \STATIC{}?)? <setterSignature>
-  \alt \EXTERNAL{}? <operatorSignature>
   \alt (\EXTERNAL{} \STATIC{}?)? <functionSignature>
-  \alt \STATIC{} \LATE{}? \FINAL{} <type>? <staticFinalDeclarationList>
-  \alt \STATIC{} \LATE{}? (\VAR{} | <type>) <initializedIdentifierList>
-  \alt \STATIC{} \CONST{} <type>? <staticFinalDeclarationList>
-  \alt \COVARIANT{} \LATE{}? (\VAR{} | <type>) <initializedIdentifierList>
-  \alt \LATE{}? \FINAL{} <type>? <initializedIdentifierList>
-  \alt \LATE{}? (\VAR{} | <type>) <initializedIdentifierList>
+  \alt \EXTERNAL{}? <operatorSignature>
+  \alt \STATIC{} (\FINAL{} | \CONST{}) <type>? <staticFinalDeclarationList>
+  \alt \STATIC{} \LATE{} \FINAL{} <type>? <initializedIdentifierList>
+  \alt (\STATIC{} | \COVARIANT{}) \LATE{}? <varOrType>
+       <initializedIdentifierList>
+  \alt \LATE{}? (\FINAL{} <type>? | <varOrType>) <initializedIdentifierList>
```

Comment:
I don't understand this re-organization.  Your comment is:

> The grammar rule for <declaration> contains a correction which makes a difference: It allows > for late final x; as well as late final x = 42;, which wasn't allowed previously.

but the text on the left includes ` \alt \LATE{}? \FINAL{} <type>? <initializedIdentifierList>` which permits both of those derivations.  What's happening here?

Reply:
Ok, per comments elsewhere, this is apparently about `static late final` without an initializer?  What about just splitting ` \alt \STATIC{} \LATE{}? \FINAL{} <type>? <staticFinalDeclarationList>` into

```
 \alt \STATIC{} \FINAL{} <type>? <staticFinalDeclarationList>
 \alt \STATIC{} \LATE{} \FINAL{} <type>? <initializedIdentifierList>
```

I really find these much more readable when the prefix of the alternatives reads off directly, even if it's more verbose.

Reply:
I've split them all, that is: (const | final), (static | covariant), (final type? | varOrType).